toolchain/sem_ir/function.h

```diff
@@ -109,6 +111,36 @@ struct Function : public EntityWithParamsBase,
     out << "}";
   }
 
+  // Returns the builtin function kind for this function, or None if this is not
+  // a builtin function.
+  auto builtin_function_kind() const -> BuiltinFunctionKind {
+    return special_function_kind == SpecialFunctionKind::Builtin
+               ? special_function_kind_info.builtin_function_kind
+               : BuiltinFunctionKind::None;
+  }
+
+  // Sets that this function is a builtin function.
+  auto SetBuiltinFunction(BuiltinFunctionKind kind) -> void {
```

Comment:
Why not mutator naming, like `set_builtin_function_kind`? This looks particularly odd next to `builtin_function_kind`, both with the different naming scheme but also the `_kind` difference.

That said, part of this is also that I tend to prefer keeping accessors/mutators separate from other functions -- i.e., I'd have probably placed these below `GetDeclaredReturnType` (and looking further, the static `GetParamPatternInfoFromPatternId` in the middle is already kind of odd).

Similarly for `thunk_decl_id`/`SetThunk`.

Reply:
Moved the setters to the end of the class.

The reason for the difference in naming is that I'm thinking of these `Set` functions as doing something more than just setting a property -- they're also setting the kind of the function. (And we might want more than a single field for one of these special function kinds at some point, in which case I'd want the setter to take multiple parameters.)

Reply:
Note the multiple fields thing is just a parenthetical -- the point is that these functions are notionally (and in practice) doing more than just setting a single property. I think mutator naming would be confusing for a function that does more than just setting the one field.